src/NServiceBus.Persistence.AzureTable/SagaPersisters/SagaOperations.cs

```diff
@@ -1,85 +1,93 @@
+        public SagaUpdate(TableEntityPartitionKey partitionKey, TableEntity sagaRow, TableClient tableClient) : base(partitionKey)
         {
             this.sagaRow = sagaRow;
+            this.tableClient = tableClient;
         }
 
-        public override CloudTable Apply(TableBatchOperation transactionalBatch)
+        public override TableClient Apply(List<TableTransactionAction> transactionalBatch)
         {
-            transactionalBatch.Add(TableOperation.Replace(sagaRow));
-            return sagaRow.Table;
+            transactionalBatch.Add(new TableTransactionAction(TableTransactionActionType.UpdateReplace, sagaRow));
+            return tableClient;
         }
     }
 
     class SagaDelete : Operation
     {
-        readonly DictionaryTableEntity sagaRow;
+        readonly TableEntity sagaRow;
+        readonly TableClient tableClient;
 
-        public SagaDelete(TableEntityPartitionKey partitionKey, DictionaryTableEntity sagaRow) : base(partitionKey)
+        public SagaDelete(TableEntityPartitionKey partitionKey, TableEntity sagaRow, TableClient tableClient) : base(partitionKey)
         {
             this.sagaRow = sagaRow;
+            this.tableClient = tableClient;
         }
 
-        public override CloudTable Apply(TableBatchOperation transactionalBatch)
+        public override TableClient Apply(List<TableTransactionAction> transactionalBatch)
         {
-            transactionalBatch.Add(TableOperation.Delete(sagaRow));
-            return sagaRow.Table;
+            transactionalBatch.Add(new TableTransactionAction(TableTransactionActionType.Delete, sagaRow));
+            return tableClient;
         }
     }
 
     class SagaRemoveSecondaryIndex : Operation
     {
         readonly SecondaryIndex secondaryIndices;
-        readonly CloudTable table;
+        readonly TableClient table;
 
-        public SagaRemoveSecondaryIndex(TableEntityPartitionKey partitionKey, Guid sagaId, SecondaryIndex secondaryIndices, PartitionRowKeyTuple partitionRowKeyTuple, CloudTable table) : base(partitionKey)
+        public SagaRemoveSecondaryIndex(TableEntityPartitionKey partitionKey, Guid sagaId, SecondaryIndex secondaryIndices, PartitionRowKeyTuple partitionRowKeyTuple, TableClient table) : base(partitionKey)
         {
             this.sagaId = sagaId;
             this.partitionRowKeyTuple = partitionRowKeyTuple;
             this.table = table;
             this.secondaryIndices = secondaryIndices;
         }
 
-        public override CloudTable Apply(TableBatchOperation transactionalBatch)
+        public override TableClient Apply(List<TableTransactionAction> transactionalBatch)
         {
             var e = new TableEntity
             {
-                ETag = "*"
+                ETag = ETag.All
             };
 
             partitionRowKeyTuple.Apply(e);
             secondaryIndices.InvalidateCache(partitionRowKeyTuple);
```

Comment:
Ok this special case might be the blocker for the proposal to align everything ValueTuples made earlier.